daemon/api.go

```diff
@@ -634,21 +638,29 @@ func (inst *snapInstruction) Agreed(intro, license string) bool {
 var snappyInstall = snappy.Install
 
 func (inst *snapInstruction) install() interface{} {
+	// FIXME: handle license agreement
 	flags := snappy.DoInstallGC
 	if inst.LeaveOld {
 		flags = 0
 	}
-	_, err := snappyInstall(inst.pkg, inst.Channel, flags, inst)
-	if err != nil {
-		if inst.License != nil && snappy.IsLicenseNotAccepted(err) {
-			return inst.License
-		}
-		return err
-	}
+	state := inst.overlord.StateEngine().State()
+	state.Lock()
+	chg := state.NewChange("install-snap", fmt.Sprintf("Installing %s from %s", inst.pkg, inst.Channel))
+	err := snapstate.Install(chg, inst.pkg, inst.Channel, flags)
+	state.Unlock()
 
-	// TODO: return the log
-	// also to do: commands update their output dynamically, as it changes.
-	return nil
+	inst.overlord.SnapManager().Ensure()
+	inst.overlord.SnapManager().Wait()
```

Comment:
Yes, this will be broken very soon if it's not already broken.

It sounds very strange to have Install being a synchronous task given how much it takes.

To avoid sorting that out now, though, something nasty such as the following would do. But let's please talk about what we want here instead.
```
for {
        status := chg.Status()
        if status == DoneStatus || status == ErrorStatus {
                break
        }
}
```